cmd/participation/list.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+)
+
+// Lists the channels an OSN is a member of.
+func listAllChannels(osn, tlsDir string) {
+	url := fmt.Sprintf("https://%s/participation/v1/channels", osn)
+
+	client := httpClient(tlsDir)
+	bodyBytes, err := getBodyBytes(client, url)
+	if err != nil {
+		panic(err)
+	}
+
+	err = printResponseAsJSON(bodyBytes, &channelList{}, os.Stdout)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
What about checking the status to detect a failed request, and printing the body if it is a failure?

Reply:
Refactored. Done.